docs/current_docs/api/ide-integration.mdx

```diff
@@ -214,5 +214,25 @@ When working with multiple inter-dependent Dagger modules, it can be challenging
 The TypeScript SDK automatically attaches source-maps to the module code. These are a record of the filename, the line number, and the column number that a type declaration corresponds to in the source code. The SDK analyzes the source code, and outputs these (as line comments of the form `./path/to/filename:line`) when creating the type definitions during module initialization.
 
 Most popular IDEs support source-maps, either natively or with an external plugin, such as the [Open file plugin](https://marketplace.visualstudio.com/items?itemName=Fr43nk.seito-openfile) for Visual Studio Code.
+</TabItem>
+<TabItem value="PHP">
+When opening the generated files in `dagger/src/`, most IDEs will automatically recognize the `dagger/dagger` package, so long as the `composer.json` file has a path configured on it.
```

Comment:
I'm not sure if I understood this correctly. I use Intelephense for VS Code and when I disable the extension, the type hinting and other features disappear.

Reply:
I will reword this, as you've pointed out, it only works if your LSP (Language Server Protocol) is enabled.

I may split it out the editors into several tabs like Python does.

- PhpStorm works out of the box, I believe it uses a built-in proprietary LSP.
- General purpose editors such as VSCode, Emacs, Vim or Zed require an external LSP.

The two most common LSPs for PHP:
- [Intelephense](https://github.com/bmewburn/vscode-intelephense/issues) (freemium)
  - I have not tested this LSP
- [Phpactor](https://github.com/phpactor/phpactor) (open source)
  - This works as long as your dagger module is within the scope of a git repo. Otherwise it struggles to locate the sdk. 

I should be able to get to this at some point in the next 2 days.